Commenting igakit to support python3.12

campospinto opened the pull request:
This PR disables all the imports to igakit, as well as all the dependent psydac functions, so that users can install and run psydac with python3.12 (see psydac issues [406](https://github.com/pyccel/psydac/issues/406) and [53](https://github.com/pyccel/psydac/issues/53) )

This temporarily disables nurbs in psydac: In a further PR the nurbs functions should re-implemented, or re-imported from a stable source.

Some changes are also made in the continuous integration workflow:
- use of a development version of sympde to allow for more flexible development
- more robust install of h5py to avoid issues with github cache

Finally some numpy typed mapping parameters are converted to native python types to avoid errors with sympify

campospinto committed:
commenting imports to igakit

campospinto committed:
update supported python version

campospinto committed:
Update continuous-integration.yml

campospinto committed:
fixing minor bug in example

campospinto committed:
Update pyproject.toml to use latest sympde version

campospinto committed:
Update pyproject.toml to use latest sympde version

campospinto committed:
commenting sympde in pyproject.toml

campospinto committed:
install dev version of sympde in continuous-integration.yml

campospinto committed:
correct  continuous-integration.yml

campospinto committed:
debugging helmoltz example

campospinto committed:
fix failing tests

campospinto committed:
Merge branch 'devel' into commenting_igakit

campospinto committed:
comment non-working sympde test

campospinto committed:
convert integral to native python to avoid sympify error

campospinto committed:
marking xfail test with strange error

campospinto committed:
fixing numpy and sympy sqrt in test

campospinto committed:
mark xfail nurbs geometry tests

campospinto committed:
try not restricting Cython version

campospinto committed:
Update continuous-integration.yml

campospinto committed:
Update continuous-integration.yml

campospinto committed:
Update continuous-integration.yml

campospinto committed:
Update continuous-integration.yml

campospinto committed:
Update continuous-integration.yml

campospinto committed:
restrict back Cython < 3.0

campospinto committed:
temp comment of serial tests

campospinto committed:
put back serial tests

campospinto committed:
export HDF5_MPI="ON" on separate line

saidctb commented:
You don't have to comment all parts that uses igakit, you only need to comment the tests that uses it, i.e. `psydac/cad/tests/test_geometry.py`

campospinto committed:
install non editable sympde in CI

kvrigor committed:
CI: Installed sympde-dev after h5py install

This ensures h5py with MPI support is installed instead
of serial h5py (which is installed when installing sympde)

kvrigor committed:
CI: Ensure that parallel h5py is installed

kvrigor committed:
CI: Download and install sympde-dev under /tmp

kvrigor committed:
CI: Comments on parallel h5py test

campospinto committed:
commenting failing sympde assert

campospinto commented:
[> igakit](https://github.com/campospinto/psydac_dev/pull/9#issuecomment-2243798022)

What do you mean ? We cannot install igakit anymore, so how can we have a running code with some parts still using igakit ?

campospinto committed:
using specific sympde branch for testing

campospinto committed:
putting back a sympde verification

campospinto committed:
put back old expected solution norm

campospinto committed:
reverting tests after sympde revert

campospinto committed:
unmarking test as xfail

campospinto committed:
use master version of sympde in continuous-integration.yml

e-moral-sanchez commented on mesh/generate_pipe.py:
Maybe we can add a `NotImplementedError` to make sure this file is not called.

campospinto replied:
indeed

e-moral-sanchez commented on mesh/multipatch/create_magnet.py:
Same comment with the `NotImplementedError`

campospinto replied:
same answer :)

e-moral-sanchez commented on psydac/api/tests/test_2d_complex.py:
```diff
@@ -487,59 +529,66 @@ def teardown_function():
-    )
+        from collections import OrderedDict
+
+        from sympy       import lambdify
```
Already imported above

campospinto replied:
I removed it

e-moral-sanchez commented on psydac/api/tests/test_2d_navier_stokes.py:
```diff
@@ -362,22 +362,25 @@ def test_st_navier_stokes_2d():
 
+    print('L2_error_norm(u) = {}'.format(l2_error_u))
+    print('L2_error_norm(p) = {}'.format(l2_error_p))
+
```
This will be printed every time the tests run and probably slows them. Maybe we should add a "debug" flag that activates all the prints.

campospinto replied:
sorry I thought that pytest just disregards print statements but it's more complicated than that. I removed these prints which are not needed

e-moral-sanchez commented on psydac/api/tests/test_assembly.py:
```diff
@@ -532,6 +531,8 @@ def test_assembly_no_synchr_args(backend):
 #==============================================================================
 if __name__ == '__main__':
+    test_Norm_complex(None)
+    exit()
```
With this `exit()` the rest of the tests are not run.

campospinto replied:
this is only executed when the file is run directly (for debugging I presume), not in the tests.

e-moral-sanchez commented on psydac/cad/cad.py:
```diff
@@ -46,68 +46,71 @@ def elevate(mapping, axis, times):
 
-        else:
-            field.coeffs[idx_to] = nrb.points[idx_from]
+    # try:
```
With the `NotImplementedError` it is not necessary to comment the rest of the lines.

campospinto replied:
Ok I'll try this

e-moral-sanchez commented on psydac/cad/tests/test_geometry.py:
```diff
@@ -55,6 +55,7 @@ def test_geometry_2d_1():
     geo_1.export('geo_1.h5')
 
 #==============================================================================
+@pytest.mark.xfail  # igakit no longer supported
```
I think that `@pytest.mark.skip(reason='IgaKit not supported')` is better because then we do not have to run the tests.

campospinto replied:
indeed, seems better 👍

e-moral-sanchez commented on pyproject.toml:
```diff
@@ -7,7 +7,7 @@ name            = "psydac"
 description     = "Python package for isogeometric analysis (IGA)"
 readme          = "README.md"
-requires-python = ">= 3.8, < 3.12"
+requires-python = ">= 3.9" #, < 3.12"
```
I think it is better to set an upper bound, because we should guarantee that the versions specified here do work.

campospinto replied:
indeed 👍

e-moral-sanchez reviewed (commented):
I added some remarks, but nothing strictly necessary.

campospinto committed:
use upper bound python <= 3.12

campospinto committed:
uncommenting non-executed code

campospinto committed:
skip geometry tests using igakit

campospinto committed:
minor cleaning in tests

campospinto committed:
fix typo in formula

campospinto reviewed (commented):
Thanks for the review! I did most of the changes, let's see if the tests still pass

campospinto committed:
fix typo in pyproject.toml

campospinto committed:
allow any python 3.12.* versions

e-moral-sanchez reviewed (approved):
Thanks for the changes!

saidctb commented on psydac/mapping/tests/test_discrete_mapping.py:
```diff
@@ -267,42 +267,42 @@ def test_parallel_jacobians_irregular(geometry, npts_irregular):
-def test_nurbs_circle():
-    rmin, rmax = 0.2, 1
-    c1, c2 = 0, 0
+# def test_nurbs_circle():
```
you can mark this test as failing using pytest without the need to comment it

campospinto replied:
indeed, I missed this one 👍

campospinto committed:
skip igakit test rather than commenting it
